### DISCUSSION TITLE
nga-msi setup instructions

### DISCUSSION THREAD

**Discussion by dsm133** [ngageoint/mage-server#130] [Q&A]
Hi,

I'm interested in viewing the nga-msi plugin from the development branch, but I do not see it visible anywhere after building and installing the application.  Is there instructions somewhere I may be overlooking which enables it or further sets it up?  Thanks in advance for your time.

**Comment by restjohn** [comment]
There are no written instructions yet. The top-level `package.json`'s [`build` script](https://github.com/ngageoint/mage-server/blob/develop/package.json#L9) includes the instruction to build the [MSI plugin](https://github.com/ngageoint/mage-server/tree/develop/plugins/nga-msi).  The package in the `instance` folder, under the top-level of the repository, [references](https://github.com/ngageoint/mage-server/blob/develop/instance/package.json#L30) the MSI plugin.  The `instance` package is an example of how to assemble the MAGE Server components into a running deployment.  Written documentation will be available eventually, probably shortly after a release is finalized.

**Reply by restjohn** [reply]
Additionally, once you are running the web app, you can access the Feeds admin page by logging into your server with an administrator account, clicking on the gear icon at the upper-right to access the admin console, then clicking the _Feeds_ tab in the vertical tab bar on the left.  From there you'll be able to setup a Feed of MSI data which you will then need to add to an Event.  Additionally, ensure that your user account is in the Event, which you can do on the _Events_ admin tab.  You can then return to the map view by clicking the _MAGE_ title on the left side of the top app bar.  On the map view, ensure you are viewing the Event that has the Feed you created.  Change your Event by clicking the _Filter_ button on the top app bar.

**Comment by doeldurieux** [comment]
I just installed MAGE and need to log in as an administrator. where do I find the initial admin user and password?

**Reply by newmanw** [reply]
You should be prompted to setup on first access via web.  Are you running a specific release, if not what branch?

**Reply by doeldurieux** [reply]
mage-server:6.2.9

**Reply by doeldurieux** [reply]
i was not prompted 
Did I do something wrong?

**Reply by newmanw** [reply]
Hmmmm not sure, just built and deployed 6.2.9 and I am not seeing your issue.  What are you seeing when you navigate to the root url in the browser?

**Reply by doeldurieux** [reply]
![image](https://github.com/ngageoint/mage-server/assets/166057356/80e9539d-bd36-4854-a99d-554d49496888)

**Reply by doeldurieux** [reply]
Thank you Billy for your help. I am going to start from scratch again.

**Reply by restjohn** [reply]
Ensure you're launching MAGE with an empty MongoDB database.  If you're running the server with a database that was previously populated, the web app will not prompt you with the initial admin user setup.  You can test this by changing the database URL MAGE uses to an uninitialized database, either with a `mage.service` script argument or environment variable.  For example, `mage.service --mongo.url mongodb://127.0.0.1:27017/mage_empty_test` or `MAGE_MONGO_URL="mongodb://127.0.0.1:27017/mage_empty_test"`.

**Reply by doeldurieux** [reply]
thank you

**Reply by doeldurieux** [reply]
I got this working now but now is asking me for a UID and I don't know what information to enter there. Can anyone help please?

**Reply by newmanw** [reply]
UID can be whatever you want.  This is used only on web signin when a device id is required (can set this up later in admin settings).

**Reply by doeldurieux** [reply]
Thank you so much for the help on this I got everything working.
